include/seastar/core/seastar.hh

```diff
@@ -60,8 +60,11 @@
 #include <filesystem>
 #include <optional>
 #include <string_view>
+#include <grp.h>
 #endif
 
+struct group;
```

Comment:
this is not necessary.

the `group` struct is only used once in this header. in the following function declaration:
```c++
future<std::optional<struct group>> getgrnam(std::string_view name, char *buf, size_t buflen);
```

in this context, `struct group` serves as a forward declaration, making a separate forward declaration unnecessary.

but we do need to have 
```diff
diff --git a/src/seastar.cc b/src/seastar.cc
index ba3a1d44..6ffdb9b4 100644
--- a/src/seastar.cc
+++ b/src/seastar.cc
@@ -142,6 +142,7 @@ module;
 #include <sys/un.h>
 #include <execinfo.h>
 #include <fcntl.h>
+#include <grp.h>
 #include <malloc.h>
 #include <pthread.h>
 #include <setjmp.h>
```

otherwise when compiling the seastar C++ module, we'd have following build failure:
```
/home/kefu/.local/bin/clang++ -DFMT_SHARED -DSEASTAR_API_LEVEL=7 -DSEASTAR_DEFERRED_ACTION_REQUIRE_NOEXCEPT -DSEASTAR_HAS_MEMBARRIER -DSEASTAR_HAVE_ASAN_FIBER_SUPPORT -DSEASTAR_HAVE_HWLOC -DSEASTAR_HAVE_NUMA -DSEASTAR_HAVE_SYSTEMTAP_SDT -DSEASTAR_HAVE_URING -DSEASTAR_MODULE -DSEASTAR_SCHEDULING_GROUPS_COUNT=16 -DSEASTAR_SSTRING -Dseastar_module_EXPORTS -I/home/kefu/dev/seastar/include -I/home/kefu/dev/seastar/build/debug/gen/include -I/home/kefu/dev/seastar/src -g -std=c++23 -fPIC -U_FORTIFY_SOURCE -Wno-include-angled-in-module-purview -MD -MT src/CMakeFiles/seastar-module.dir/util/file.cc.o -MF src/CMakeFiles/seastar-module.dir/util/file.cc.o.d @src/CMakeFiles/seastar-module.dir/util/file.cc.o.modmap -o src/CMakeFiles/seastar-module.dir/util/file.cc.o -c /home/kefu/dev/seastar/src/util/file.cc
In file included from /home/kefu/dev/seastar/src/util/file.cc:28:
In file included from /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/deque:62:
In file included from /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/stl_algobase.h:64:
In file included from /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/stl_pair.h:60:
/usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/type_traits:3413:7: error: incomplete type 'seastar::group' used in type trait expression
 3413 |     = __is_trivially_constructible(_Tp, __add_lval_ref_t<const _Tp>);
      |       ^
/usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/optional:506:12: note: in instantiation of variable template specialization 'std::is_trivially_copy_constructible_v<seastar::group>' requested here
  506 |            bool = is_trivially_copy_constructible_v<_Tp>,
      |                   ^
/usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/optional:704:15: note: in instantiation of default argument for '_Optional_base<seastar::group>' required here
  704 |     : private _Optional_base<_Tp>,
      |               ^~~~~~~~~~~~~~~~~~~
/usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/type_traits:3435:63: note: in instantiation of template class 'std::optional<seastar::group>' requested here
 3435 |   requires (!is_reference_v<_Tp>) && requires (_Tp& __t) { __t.~_Tp(); }
      |                                                               ^
/usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/type_traits:3435:60: note: in instantiation of requirement here
 3435 |   requires (!is_reference_v<_Tp>) && requires (_Tp& __t) { __t.~_Tp(); }
      |                                                            ^~~~~~~~~~
/usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/type_traits:3435:38: note: while substituting template arguments into constraint expression here
 3435 |   requires (!is_reference_v<_Tp>) && requires (_Tp& __t) { __t.~_Tp(); }
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/kefu/dev/seastar/include/seastar/core/future.hh:354:15: note: (skipping 1 context in backtrace; use -ftemplate-backtrace-limit=0 to see all)
  354 |         (std::is_trivially_destructible_v<T> && std::is_trivially_constructible_v<T> &&
      |               ^
/home/kefu/dev/seastar/include/seastar/core/future.hh:354:15: note: during template argument deduction for variable template partial specialization 'is_trivially_destructible_v<_Tp>' [with _Tp = std::optional<seastar::group>]
/home/kefu/dev/seastar/include/seastar/core/future.hh:361:44: note: in instantiation of variable template specialization 'seastar::internal::can_inherit<std::optional<seastar::group>>' requested here
  361 |     : public uninitialized_wrapper_base<T, can_inherit<T>> {};
      |                                            ^
/home/kefu/dev/seastar/include/seastar/core/future.hh:571:58: note: in instantiation of template class 'seastar::internal::uninitialized_wrapper<std::optional<seastar::group>>' requested here
  571 | struct future_state :  public future_state_base, private internal::uninitialized_wrapper<T> {
      |                                                          ^
/home/kefu/dev/seastar/include/seastar/core/future.hh:1242:18: note: in instantiation of template class 'seastar::future_state<std::optional<seastar::group>>' requested here
 1242 |     future_state _state;
      |                  ^
/home/kefu/dev/seastar/src/util/file.cc:136:37: note: in instantiation of template class 'seastar::future<std::optional<seastar::group>>' requested here
  136 | future<std::optional<struct group>> getgrnam(std::string_view name, char *buf, size_t buflen) {
      |                                     ^
/home/kefu/dev/seastar/include/seastar/core/seastar.hh:373:29: note: forward declaration of 'seastar::group'
  373 | future<std::optional<struct group>> getgrnam(std::string_view name, char *buf, size_t buflen);
      |                             ^
In file included from /home/kefu/dev/seastar/src/util/file.cc:28:
```

Reply:
Ok thanks, I noticed I also had to remove the `#include <grp.h>` from the `seastar/include/seastar/core/reactor.hh` or else the compile failure was 

```
...
...
In file included from /home/raj/db/scylla/seastar/include/seastar/core/reactor.hh:77:
/usr/include/grp.h:165:11: error: reference to 'group' is ambiguous
  165 |                         struct group *__restrict __resultbuf,
      |                                ^
/usr/include/grp.h:42:8: note: candidate found by name lookup is 'group'
   42 | struct group
      |        ^
/home/raj/db/scylla/seastar/include/seastar/core/seastar.hh:370:29: note: candidate found by name lookup is 'seastar::group'
  370 | future<std::optional<struct group>> getgrnam(std::string_view name, char *buf, size_t buflen);
      |                             ^
In file included from /home/raj/db/scylla/seastar/tests/perf/future_util_perf.cc:27:
In file included from /home/raj/db/scylla/seastar/include/seastar/coroutine/parallel_for_each.hh:30:
In file included from /home/raj/db/scylla/seastar/include/seastar/core/reactor.hh:77:
/usr/include/grp.h:167:11: error: reference to 'group' is ambiguous
  167 |                         struct group **__restrict __result)
      |                                ^
/usr/include/grp.h:42:8: note: candidate found by name lookup is 'group'
   42 | struct group
      |        ^
/home/raj/db/scylla/seastar/include/seastar/core/seastar.hh:370:29: note: candidate found by name lookup is 'seastar::group'
  370 | future<std::optional<struct group>> getgrnam(std::string_view name, char *buf, size_t buflen);
      |                             ^
13 errors generated.

```